function.go

```diff
@@ -16,7 +16,7 @@ type Function struct {
 }
 
 // Call this JavaScript function with the given arguments.
-func (fn *Function) Call(args ...Valuer) (*Value, error) {
+func (fn *Function) Call(recv *Object, args ...Valuer) (*Value, error) {
```

Comment:
[v8::Function::Call](https://v8.github.io/api/head/classv8_1_1Function.html#ad916775f129a95ed93e4c68cd3848b8b) uses `Local<Value>` as the receiver type, which we should do for parity

```suggestion
func (fn *Function) Call(recv *Value, args ...Valuer) (*Value, error) {
```

as well as to support passing in an undefined value for top-level function calls, which probably makes more sense when updating existing calls

```diff
diff --git a/function_template_test.go b/function_template_test.go
index b2fed39..1f9dca1 100644
--- a/function_template_test.go
+++ b/function_template_test.go
@@ -71,7 +71,7 @@ func TestFunctionTemplateGetFunction(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	ret, err := fn.Call(ctx.Global(), ten)
+	ret, err := fn.Call(v8go.Undefined(iso), ten)
 	if err != nil {
 		t.Fatal(err)
 	}
diff --git a/function_test.go b/function_test.go
index 4bf1040..6c635f6 100644
--- a/function_test.go
+++ b/function_test.go
@@ -28,7 +28,7 @@ func TestFunctionCall(t *testing.T) {
 	failIf(t, err)
 
 	fn, _ := addValue.AsFunction()
-	resultValue, err := fn.Call(ctx.Global(), arg1, arg1)
+	resultValue, err := fn.Call(v8go.Undefined(iso), arg1, arg1)
 	failIf(t, err)
 
 	if resultValue.Int32() != 2 {
@@ -91,7 +91,7 @@ func TestFunctionCallToGoFunc(t *testing.T) {
 	failIf(t, err)
 	fn, err := val.AsFunction()
 	failIf(t, err)
-	resultValue, err := fn.Call(ctx.Global())
+	resultValue, err := fn.Call(v8go.Undefined(iso))
 	failIf(t, err)
 
 	if !called {
@@ -107,7 +107,8 @@ func TestFunctionCallError(t *testing.T) {
 
 	ctx, err := v8go.NewContext()
 	failIf(t, err)
-	defer ctx.Isolate().Dispose()
+	iso := ctx.Isolate()
+	defer iso.Dispose()
 	defer ctx.Close()
 
 	_, err = ctx.RunScript("function throws() { throw 'error'; }", "script.js")
@@ -116,7 +117,7 @@ func TestFunctionCallError(t *testing.T) {
 	failIf(t, err)
 
 	fn, _ := addValue.AsFunction()
-	_, err = fn.Call(ctx.Global())
+	_, err = fn.Call(v8go.Undefined(iso))
 	if err == nil {
 		t.Errorf("expected an error, got none")
 	}
```